### ISSUE TITLE
Clothing items with dashes in names don't work

### ISSUE THREAD

**Issue by bluecarrot16** [sanderfrenken/Universal-LPC-Spritesheet-Character-Generator#10]
I can't select items (mostly pirate hat styles) that have a dash in the variant name; e.g. Tricorne lieutenant, which has variants named "black-bronze", "black-gold", "black-silver", etc. Instead, when I click on the item once, nothing happens. I would expect the URL to become <https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/#?body=Humanlike_white&hat=Tricorne_lieutenant_black-bronze>

Weirdly, if I click on the item multiple times, it eventually works, but the URL gets gradually more messed up, and the selected object disappears if I refresh the page.
- Click 2: <https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/#?body=Humanlike_white&hat=Tricorne_lieutenant_black-bronze&hat-Tricorne_lieutenant_black=bronze>
- Click 3: <https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/#?body=Humanlike_white&hat=Tricorne_lieutenant_black-bronze&hat-tricorne_lieutenant_black=bronze&hat-Tricorne_lieutenant_black=bronze>

Do you know why this is happening? Are dashes prohibited? I noticed after I had added all the pirate stuff that other objects have underscores rather than dashes in their names, so we can maybe fix this by changing all the dashes to underscores in the files, variant names, etc...

**Comment by sanderfrenken** [comment]
hee there, indeed dashes are prohibited as they are used in the JS for interpreting the query params:

https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/blob/6870365d61b376247b6e66daa457e0dcbbac6068/sources/chargen.js#L459

The variants could have underscores, EG: 


![](https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/blob/master/spritesheets/hair/bangslong/female/dark_blonde.png)

Then in the sheet definitions, you can use a SPACE to make the field readable for the user:

https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/blob/6870365d61b376247b6e66daa457e0dcbbac6068/sheet_definitions/hair_bangslong.json#L21

This is because in the template here:

https://github.com/sanderfrenken/Universal-LPC-Spritesheet-Character-Generator/blob/6870365d61b376247b6e66daa457e0dcbbac6068/sources/chargen_templates.js#L38

white space is replaced with an underscore to get the actual sprite sheet reference.

So in a nutshell, yes dashes are prohibited, and yes they should be changed to underscores for the name of the actual sprite sheets. In the variant names, you can replace the _ with a space to make it more userfriendly readable.

Does that answer your question?
